vite.config.js

```diff
@@ -0,0 +1,55 @@
+import { defineConfig } from "vite";
+import pkg from "./package.json" with { type: "json" };
+
+const indexInput = "src/index.ts";
+const tokenInput = "src/token.ts";
+const globalVarName = pkg.name;
+
+export default defineConfig(({ mode }) => {
+  const isCJSBuild = mode === "production";
+
+  return {
+    build: {
+      // for UMD build we do not want to empty directory, so previous builds stay intact
+      emptyOutDir: isCJSBuild,
+      minify: true,
+      sourcemap: true,
+      // UMD build targets lower level ES, while CJS the lowest Node.js LTS version
+      target: isCJSBuild ? "es2022" : "es6",
+      lib: {
+        // leave out token from UMD build
+        entry: isCJSBuild ? [indexInput, tokenInput] : indexInput,
+        name: isCJSBuild ? undefined : globalVarName,
+        formats: isCJSBuild ? ["cjs"] : ["umd"],
+        fileName: (format, entryName) => {
+          switch (format) {
+            case "umd":
+              return `umd/${entryName}.min.js`;
+            case "cjs":
+              return `cjs/${entryName}.cjs`;
+            default:
+              throw new Error(`unsupported format ${format}`);
+          }
+        },
+      },
+      rollupOptions: isCJSBuild
+        ? {
+            // make sure external imports that should not be bundled are listed here for CJS build
+            external: ["node:crypto"],
+          }
+        : // https://github.com/vitejs/vite/issues/11624
```

Comment:
Thanks for figuring that out.

Can you please add two comment?
- One comment explaining what this code does, e.g. enable the window.Meilisearch syntax
- One TODO comment so we know where to look to remove this in the future

Reply:
Did it!